server/src/main/java/org/elasticsearch/index/mapper/GeoShapeQueryable.java

```diff
@@ -8,28 +8,215 @@
 
 package org.elasticsearch.index.mapper;
 
+import org.apache.lucene.geo.GeoEncodingUtils;
+import org.apache.lucene.geo.LatLonGeometry;
+import org.apache.lucene.search.MatchNoDocsQuery;
 import org.apache.lucene.search.Query;
+import org.elasticsearch.common.geo.GeometryNormalizer;
+import org.elasticsearch.common.geo.Orientation;
 import org.elasticsearch.common.geo.ShapeRelation;
 import org.elasticsearch.common.geo.SpatialStrategy;
+import org.elasticsearch.geometry.Circle;
 import org.elasticsearch.geometry.Geometry;
+import org.elasticsearch.geometry.GeometryCollection;
+import org.elasticsearch.geometry.GeometryVisitor;
+import org.elasticsearch.geometry.Line;
+import org.elasticsearch.geometry.LinearRing;
+import org.elasticsearch.geometry.MultiLine;
+import org.elasticsearch.geometry.MultiPoint;
+import org.elasticsearch.geometry.MultiPolygon;
+import org.elasticsearch.geometry.Point;
+import org.elasticsearch.geometry.Polygon;
+import org.elasticsearch.geometry.Rectangle;
+import org.elasticsearch.index.query.QueryShardException;
 import org.elasticsearch.index.query.SearchExecutionContext;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 /**
  * Implemented by {@link org.elasticsearch.index.mapper.MappedFieldType} that support
  * GeoShape queries.
  */
 public interface GeoShapeQueryable {
 
-    Query geoShapeQuery(Geometry shape, String fieldName, ShapeRelation relation, SearchExecutionContext context);
+    Query geoShapeQuery(SearchExecutionContext context, String fieldName, ShapeRelation relation, LatLonGeometry... luceneGeometries);
+
+    default Query geoShapeQuery(SearchExecutionContext context, String fieldName, ShapeRelation relation, Geometry shape) {
+        final LatLonGeometry[] luceneGeometries = toQuantizeLuceneGeometry(fieldName, context, shape, relation);
+        if (luceneGeometries.length == 0) {
+            return new MatchNoDocsQuery();
+        }
+        return geoShapeQuery(context, fieldName, relation, luceneGeometries);
+    }
 
     @Deprecated
     default Query geoShapeQuery(
```

Comment:
This function can be deleted. It is the same as the function it calls, but with an extra field that is always ignored.

Reply:
See `LegacyGeoShapeFieldMapper` where the parameter SpatialStrategy is needed.  This is a legacy parameter that we need to carry over until we remove the old way of working with geo_shape field.

Reply:
Except the parameter is completely ignored in the implementation, so it can also be deleted from `LegacyGeoShapeFieldMapper`, right?